src/account.ts

```diff
@@ -373,6 +374,19 @@ export class Account {
         return result.result && result.result.length > 0 && parse(Buffer.from(result.result));
     }
 
+    async viewState(prefix: string | Uint8Array, blockQuery: { blockId: BlockId } | { finality: Finality } ) {
+        const { blockId, finality } = blockQuery as any || {};
+        const { values } = await this.connection.provider.query({
+            request_type: 'view_state', 
+            block_id: blockId,
+            finality: blockId ? undefined : finality || 'optimistic',
```

Comment:
There's a pretty tricky interplay between `blockId` & `finality` here that I'd like to see documented more explicitly. Can we add a code comment above `viewState` to get picked up by typedoc, as well as throwing an error message when you pass a nonsensical combination of the two?

Reply:
yeah, I forgot about the doc here, need to add

Reply:
> as well as throwing an error message when you pass a nonsensical combination of the two?

Types kinda describe it. We aren't really checking types during runtime in most other places. TBH this is something I really don't understand about TypeScript. Like doing this whole effort with types and then not having easy way to check in runtime automatically 🤷‍♂️

Reply:
created issue to track https://github.com/near/near-api-js/issues/457